src/main/java/com/google/sps/storage/StorageControllerInterface.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.storage;
+
+/**
+ * An interface for providing unified I/O to the storage component.
+ */
+public interface StorageControllerInterface {
+  /**
+   * Creates an entirely new database entry, with just the metadata of a commit,
+   * having the passed/failed fields empty.
+   *
+   * @param entryData a JSON in plain string format, must have a "commitHash" field.
+   * @throws IllegalArgumentException if entryData is null or doesn't have a
+   *      "commitHash" field.
+   */
+  void createNewEntry(final String entryData) throws IllegalArgumentException;
```

Comment:
I'd recommend to not declare the `IllegalArgumentException` as checked here, but leave it unchecked (remove `throws IllegalArgumentException`) (still keep it in javadocs though). The reasoning is that the cases where this method would throw this exception are indications of programmer errors (the user of the code passed an inappropriate value) and programmer errors are usually handled as unchecked exceptions in Java as far as I know.